pkg/util/validation/limits.go

```diff
@@ -118,6 +119,10 @@ type Limits struct {
 	QueryShardingTotalShards       int            `yaml:"query_sharding_total_shards" json:"query_sharding_total_shards"`
 	QueryShardingMaxShardedQueries int            `yaml:"query_sharding_max_sharded_queries" json:"query_sharding_max_sharded_queries"`
 	SplitInstantQueriesByInterval  model.Duration `yaml:"split_instant_queries_by_interval" json:"split_instant_queries_by_interval" category:"experimental"`
+
+	// Query-frontend limits.
+	MaxTotalQueryLength model.Duration `yaml:"max_total_query_length,omitempty" json:"max_total_query_length,omitempty"`
```

Comment:
I don't feel strong about it, but I would mark `category:"experimental"` for now. If so, please add it to the list of experimental features in `docs/sources/operators-guide/configure/about-versioning.md`.